src/lib.rs

```diff
@@ -414,61 +415,70 @@ impl ClientSubsetSync for RpcClient {
-                    preflight_commitment: None,
-                    encoding: None,
-                    max_retries: None,
-                },
-            ) {
-                Ok(sig) => break sig,
-                Err(client_error) => {
-                    if matches!(client_error.kind, ClientErrorKind::TransactionError(_)) {
-                        let encoded_tx = match self.get_transaction_with_config(
-                            &signature,
-                            RpcTransactionConfig {
-                                encoding: Some(UiTransactionEncoding::JsonParsed),
-                                commitment: Some(CommitmentConfig::confirmed()),
-                                max_supported_transaction_version: None,
-                            },
-                        ) {
-                            Ok(tx) => tx,
-                            Err(e) => {
-                                // This is most likely an RPC issue
-                                // TODO: implmement retry/backoff logic
-                                return Err(EllipsisClientError::from(anyhow::Error::msg(
-                                    format!("Failed to fetch transaction ({}): {}", signature, e),
-                                )));
-                            }
-                        };
-                        let logs = parse_transaction(encoded_tx).logs;
-                        return Err(EllipsisClientError::TransactionFailed { signature, logs });
-                    } else {
-                        retries += 1;
-                        println!(
-                            "Attempt {} failed: Resending transaction ({}). Error: {}",
-                            retries, signature, client_error
-                        );
-                        if retries == 3 {
-                            return Err(EllipsisClientError::from(anyhow::Error::msg(format!(
-                                "Failed to send transaction ({}): {}",
-                                signature, client_error
-                            ))));
-                        }
-                        continue;
-                    }
+        let blockhash = self
+            .get_latest_blockhash_with_commitment(CommitmentConfig::confirmed())?
+            .0;
+        tx.partial_sign(&signers.to_vec(), blockhash);
+        let signature = self.send_transaction_with_config(
+            &tx,
+            RpcSendTransactionConfig {
+                skip_preflight: false,
+                preflight_commitment: Some(CommitmentLevel::Processed),
+                encoding: None,
+                max_retries: None,
+                min_context_slot: None,
+            },
+        )?;
+
+        let (signature, status) = loop {
+            // Get recent commitment in order to count confirmations for successful transactions
+            let status = self
+                .get_signature_status_with_commitment(&signature, CommitmentConfig::processed())?;
```

Comment:
processed?

Reply:
this first step makes sure that someone received our tx